cleanlab_studio/internal/tlm/validation.py

```diff
@@ -23,16 +23,18 @@
 
 def validate_tlm_prompt_kwargs_constrain_outputs(
     prompt: Union[str, Sequence[str]],
+    valid_kwargs_check: bool = True,
     **kwargs: Any,
 ) -> Union[Optional[List[str]], Optional[List[Optional[List[str]]]]]:
-    # validate kwargs - only allow constrain_outputs
-    supported_kwargs = TLM_VALID_KWARGS
-    unsupported_kwargs = set(kwargs.keys()) - supported_kwargs
-    if unsupported_kwargs:
-        raise ValidationError(
-            f"Unsupported keyword arguments: {unsupported_kwargs}. "
-            f"Supported keyword arguments are: {supported_kwargs}"
-        )
+    if valid_kwargs_check:
+        # validate kwargs - only allow constrain_outputs
+        supported_kwargs = TLM_VALID_KWARGS
+        unsupported_kwargs = set(kwargs.keys()) - supported_kwargs
+        if unsupported_kwargs:
+            raise ValidationError(
+                f"Unsupported keyword arguments: {unsupported_kwargs}. "
+                f"Supported keyword arguments are: {supported_kwargs}"
+            )
```

Comment:
I think we should separate this check helper function into a few separate pieces - one to check if the kwargs are valid, and then another one to specifically validate what users pass into `constrain_outputs`. That second function can then be reused for both `prompt` and `score` in a cleaner way